src/FSharp.Control.TaskSeq/TaskSeqInternal.fs

```diff
@@ -613,6 +631,103 @@ module internal TaskSeqInternal =
                     | false -> ()
         }
 
+
+    let skipOrTake skipOrTake count (source: TaskSeq<_>) =
+        checkNonNull (nameof source) source
+
+        if count < 0 then
+            raiseCannotBeNegative (nameof count)
+
+        match skipOrTake with
+        | Skip ->
+            // don't create a new sequence if count = 0
+            if count = 0 then
+                source
+            else
+                taskSeq {
+                    use e = source.GetAsyncEnumerator CancellationToken.None
+
+                    for _ in 1..count do
+                        let! step = e.MoveNextAsync()
+
+                        if not step then
+                            raiseInsufficient ()
+
+                    while! e.MoveNextAsync() do
+                        yield e.Current
+
+                }
+        | Drop ->
+            // don't create a new sequence if count = 0
+            if count = 0 then
+                source
+            else
+                taskSeq {
+                    use e = source.GetAsyncEnumerator CancellationToken.None
+
+                    let! step = e.MoveNextAsync()
+                    let mutable cont = step
+                    let mutable pos = 0
+
+                    // skip, or stop looping if we reached the end
+                    while cont do
+                        pos <- pos + 1
+
+                        if pos < count then
+                            let! moveNext = e.MoveNextAsync()
+                            cont <- moveNext
+                        else
+                            cont <- false
+
+                    // return the rest
+                    while! e.MoveNextAsync() do
+                        yield e.Current
+
+                }
+        | Take ->
+            // don't initialize an empty task sequence
+            if count = 0 then
+                empty
+            else
+                taskSeq {
+                    use e = source.GetAsyncEnumerator CancellationToken.None
+
+                    for _ in count .. - 1 .. 1 do
+                        let! step = e.MoveNextAsync()
+
+                        if not step then
+                            raiseInsufficient ()
+
+                        yield e.Current
+                }
+
+        | Truncate ->
+            // don't create a new sequence if count = 0
+            if count = 0 then
+                empty
+            else
+                taskSeq {
+                    use e = source.GetAsyncEnumerator CancellationToken.None
+
+                    let! step = e.MoveNextAsync()
+                    let mutable cont = step
+                    let mutable pos = 0
+
+                    // return items until we've exhausted the seq
+                    // report this line, weird error:
```

Comment:
todo?

Reply:
Yeah, I haven't reported that line yet. It is the most confusing error I've ever seen 😆. I'll remove that line before merge. Tx for the review!